File: solutions/csharp/bob/1/Bob.cs
using System;

public static class Bob
{
    public static bool IsUpper (this String str) =>  str == str.ToUpper() && (str.Contains('A') ||str.Contains('E') ||str.Contains('I') || str.Contains('O')||str.Contains('U')) ? true : false;
  
    public static string Response(string statement)
    {     
        string r = "Whatever.";
       if (String.IsNullOrWhiteSpace(statement)){
           r="Fine. Be that way!";
       }else if (statement.Trim().EndsWith('?') && statement.IsUpper()){
       r= "Calm down, I know what I'm doing!";
       }else if (statement.Trim().EndsWith('?')){
       r="Sure.";
       }else if (statement.IsUpper()){
       r="Whoa, chill out!";
       }

        return r;
    }
}

File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val isYelling = input.trim().toCharArray().all{ it.isUpperCharOrOther() } && input.trim().toCharArray().any{ it.isLetter() }
        val isQuestion = input.trim().length > 0 && input.trim().last().compareTo('?') == 0
        val isSilence = input.isBlank()
        when {
            isYelling && isQuestion -> return "Calm down, I know what I'm doing!";
            isQuestion -> return "Sure.";
            isYelling -> return "Whoa, chill out!";
            isSilence -> return "Fine. Be that way!";
            else -> return "Whatever.";
        }
    }
}

fun Char.isUpperCharOrOther() = !isLetter() || isUpperCase()
